app/models/project_services/composer_service.rb

```diff
@@ -0,0 +1,464 @@
+              ['Tusk Asset', 'tusk-asset'],
+              ['Tusk Command', 'tusk-command'],
+              ['Tusk Task', 'tusk-task'],
+              ['TYPO3 CMS Extension', 'typo3-cms-extension'],
+              ['TYPO3 Flow Boilerplate', 'typo3-flow-boilerplate'],
+              ['TYPO3 Flow Build', 'typo3-flow-build'],
+              ['TYPO3 Flow Framework', 'typo3-flow-framework'],
+              ['TYPO3 Flow Package', 'typo3-flow-package'],
+              ['TYPO3 Flow Plugin', 'typo3-flow-plugin'],
+              ['TYPO3 Flow Site', 'typo3-flow-site'],
+              ['Wolf CMS Plugin', 'wolfcms-plugin'],
+              ['WordPress Core', 'wordpress-core'],
+              ['WordPress Must Use Plugin', 'wordpress-muplugin'],
+              ['WordPress Plugin', 'wordpress-plugin'],
+              ['WordPress Theme', 'wordpress-theme'],
+              ['Zend Extra', 'zend-extra'],
+              ['Zend Library', 'zend-library'],
+              ['Zend Module', 'zend-module'],
+              ['Zikula Module', 'zikula-module'],
+              ['Zikula Theme', 'zikula-theme']
+            ],
+            default_choice: 'library'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Branches:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_branches'
+          },
+          { type: 'text',
+            name: 'branch_filters',
+            placeholder: 'branches you wish to export comma separated.',
+            hint: 'Separate branches with commas. '\
+                  'Leave blank to export all branches.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Tags:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_tags'
+          },
+          { type: 'text',
+            name: 'tag_filters',
+            placeholder: 'tags you wish to export comma separated.',
+            hint: 'Separate tags with commas. '\
+                  'Leave blank to export all tags.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Advanced:', fields:
+        [
+          { type: 'textarea',
+            name: 'custom_json',
+            placeholder: 'custom json to use for exporting this package.'
+          }
+        ]
+      }
+    ]
+  end
+
+  # disable test button
+  def can_test?
+    false
+  end
+
+  def supported_events
+    %w(push tag_push)
+  end
+
+  def execute(push_data)
+    process_project
+  end
+
+  def process_project
+
+    # delete the project repository file since it will be regenerated
+    if File.exists?(repo_path)
+      File.delete(repo_path)
+    end
+
+    if activated?
+      # process packages for all tags
+      project.repository.tags.each do |tag|
+        process_commit(tag)
+      end
+
+      # process packages for all branches
+      project.repository.branches.each do |branch|
+        process_commit(branch)
+      end
+
+      repository.write
+    end
+    update_root_file
+  rescue
+    # skip on error
```

Comment:
There are two kind of errors that can occur here, namely file system errors and invalid package errors.

The first kind are file system errors that can happen due to file/dir permissions, which is something that needs to be taken care of during the installation of gitlab. Then there could be faulty storage, etc. These type of errors are critical and I would like your suggestion on how to handle them.

The second kind of errors is when we have invalid packages which can happen due to: 
1. unnamed packages, 
2. unsupported version scheme (non schematic)
3. missing/invalid root composer.json when packaging mode is set to default.

These kind of errors are beeing taken care of in the process_commit function and are non critical. Perhaps we should log these in the Application log?.

Please note that I am not an expert ruby developer so your assistance is greatly appreciated.